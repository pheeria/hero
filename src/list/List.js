import React, { useState } from 'react';
import { List as AntList, Rate, Icon } from 'antd';

const List = ({ places, selectedPlace, removePlace }) => {
  const [sortAscending, setSortAscending] = useState(true);
  const [sortedPlaces, setSortedPlaces] = useState(places);
  return (
    <div>
      <Icon
        type={sortAscending ? 'sort-ascending' : 'sort-descending'}
        onClick={() => {
          setSortAscending(!sortAscending);
          setSortedPlaces(
            sortedPlaces.sort((a, b) =>
              sortAscending ? a.name < b.name : a.name > b.name
            )
          );
        }}
      />
      <AntList
        itemLayout="horizontal"
        dataSource={sortedPlaces}
        renderItem={place => (
          <AntList.Item
            actions={[<Icon onClick={() => removePlace(place)} type="close" />]}
            style={
              selectedPlace && selectedPlace.id === place.id
                ? { border: `1px solid red` }
                : null
            }
          >
            <AntList.Item.Meta
              title={<p>{place.name}</p>}
              description={
                <div>
                  <div>
                    <Icon type="heart" theme="twoTone" twoToneColor="#eb2f96" />
                    : {place.favoriteFood}
                  </div>
                  <div>
                    <Rate disabled defaultValue={place.rating} />
                  </div>
                </div>
              }
            />
          </AntList.Item>
        )}
      />
    </div>
  );
};

export default List;
