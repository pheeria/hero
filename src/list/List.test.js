import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from './List';

describe('List', () => {
  it('should render without crashing', () => {
    const places = [
      {
        id: 'uuid',
        name: 'Some Place',
        rating: 5,
        favoriteFood: 'Et',
        location: {
          lat: 52.5248,
          lng: 13.393141600000035
        }
      },
      {
        id: 'uuid2',
        name: 'Berlin Cigkofte',
        rating: 5,
        favoriteFood: 'Cig kofte',
        location: {
          lat: 52.47732379999999,
          lng: 13.425160300000016
        }
      }
    ];
    const wrapper = shallow(
      <List
        places={places}
        selectedPlace={{}}
        removePlace={jest.fn()}
        selectPlace={jest.fn()}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
