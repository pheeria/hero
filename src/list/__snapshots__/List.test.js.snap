// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render without crashing 1`] = `
<Col>
  <Row
    align="middle"
    gutter={0}
    justify="space-between"
    style={
      Object {
        "backgroundColor": "#000010",
        "height": "64px",
        "padding": "10px",
      }
    }
    type="flex"
  >
    <Col
      span={20}
    >
      <AutoComplete
        choiceTransitionName="zoom"
        dataSource={
          Array [
            "Some Place",
            "Berlin Cigkofte",
          ]
        }
        filterOption={false}
        onSearch={[Function]}
        onSelect={[Function]}
        optionLabelProp="children"
        placeholder="Search"
        showSearch={false}
        transitionName="slide-up"
      />
    </Col>
    <Col
      span={2}
    >
      <Icon
        onClick={[Function]}
        style={
          Object {
            "color": "white",
          }
        }
        type="sort-ascending"
      />
    </Col>
  </Row>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "favoriteFood": "Et",
          "id": "uuid",
          "location": Object {
            "lat": 52.5248,
            "lng": 13.393141600000035,
          },
          "name": "Some Place",
          "rating": 5,
        },
        Object {
          "favoriteFood": "Cig kofte",
          "id": "uuid2",
          "location": Object {
            "lat": 52.47732379999999,
            "lng": 13.425160300000016,
          },
          "name": "Berlin Cigkofte",
          "rating": 5,
        },
      ]
    }
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
    style={
      Object {
        "height": "calc(100vh - 64px)",
        "overflow": "auto",
      }
    }
  />
</Col>
`;
